### PR TITLE
Refactor, Cleanup Code and use Generics

### PR DESCRIPTION
**Pull request by jabdoa2**
- Refactored handler to use generics instead of "podController" wrapper type
- Deduplicated Getter Types
- Deduplicated Controller logic and entrypoints
- Deduplicated Webhook entrypoints
- Deduplicated Controller Tests (using generics)
  - Tests are now exactly the same for all three objects (there were slightly diverged before)
  - Deduplicated podTemplate in Test Data

Overall we lost about 1.2k lines of code. Also extending/maintaining controllers and controller tests became easier. Before you basically had to make the same change in three files each. That is now in one place. Only downside is that you have to deal with generic types in tests now.

### CODE REVIEW

**Comment by jabdoa2** [comment]
I will resolve the conflict. I also found why my tests for #167 failed. I will improve the test setup to make those easier to add.

File: pkg/controller/deployment/deployment_controller_test.go [hunk 1]
```diff
@@ -16,6 +16,60 @@ limitations under the License.
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+package deployment
+
 import (
```
Comment (toelke) [inline pkg/controller/deployment/deployment_controller_test.go]: .

File: pkg/core/controller.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(source.Kind(mgr.GetCache(), typeInstance), &handler.EnqueueRequestForObject{}, predicate.Or(predicate.GenerationChangedPredicate{}, predicate.AnnotationChangedPredicate{}))
+	if err != nil {
+		return err
+	}
+
+	// Watch ConfigMaps owned by a DaemonSet
```
Comment (toelke) [inline pkg/core/controller.go:42]:
```suggestion
	// Watch ConfigMaps owned by Resource
```

File: pkg/core/controller.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	err = c.Watch(source.Kind(mgr.GetCache(), &corev1.ConfigMap{}), EnqueueRequestForWatcher(h.GetWatchedConfigmaps()))
+	if err != nil {
+		return err
+	}
+
+	// Watch Secrets owned by a DaemonSet
```
Comment (toelke) [inline pkg/core/controller.go:48]:
```suggestion
	// Watch Secrets owned by Resource
```

**Review by toelke — APPROVED** [review]

**Pull request merged**